### PR TITLE
Fix handling of weird file names in git_status plugin

### LINKED ISSUE #1385: git status plugin: ä character
1. Open a project tracked with Git in porcupine.
2. Create a file with `ä` in file name. Commit it to Git.
3. Open the file in Porcupine.
4. Modify the file and save.

Expected result: file shows up as red in Porcupine's directory tree

Actual result: file shows up as white

### PR DESCRIPTION
**Pull request by Akuli**
Fixes #1385 
ss

### CODE REVIEW

File: porcupine/plugins/git_status.py [hunk 1]
```diff
@@ -28,6 +29,47 @@
...
+# The \xxx means byte xxx specified as octal. First digit is 0-3 because the
+# biggest possible byte value is 255, which is 0o377 octal.
+#
+# Because this would be too easy, Git also special-cases some characters. For
+# example, tabs come out as \t rather than \011.
+_SPECIAL_ESCAPES = {
+    # There are probably more, but hopefully this covers everything
+    # that comes up in real-world projects
+    b"\\t": b"\t",  # \t = tab
+    b"\\n": b"\n",  # \t = newline
```
Comment (Akuli) [inline porcupine/plugins/git_status.py:44]:
```suggestion
    b"\\n": b"\n",  # \n = newline
```

File: tests/test_git_status_plugin.py [hunk 2]
```diff
@@ -29,6 +31,36 @@ def test_added_and_modified_content(tree, tmp_path, monkeypatch):
...
+        "foo<bar.txt",
+        "foo>bar.txt",
+        "foo:bar.txt",
+        'foo"bar.txt',
+        r"foo\bar.txt",
+        r"foo\123.txt",  # not a special escape code, only looks like it
+        "foo|bar.txt",
+        "foo?bar.txt",
+        "foo*bar.txt",
+        # Not mentioned in linked stackoverflow answer but still doesn't work
```
Comment (Akuli) [inline tests/test_git_status_plugin.py:47]:
```suggestion
        # Not mentioned in linked stackoverflow answer but still doesn't work on Windows
```

File: porcupine/plugins/git_status.py [hunk 3]
```diff
@@ -28,6 +29,47 @@
...
+# The \xxx means byte xxx specified as octal. First digit is 0-3 because the
+# biggest possible byte value is 255, which is 0o377 octal.
+#
+# Because this would be too easy, Git also special-cases some characters. For
+# example, tabs come out as \t rather than \011.
+_SPECIAL_ESCAPES = {
+    # There are probably more, but hopefully this covers everything
+    # that comes up in real-world projects
+    b"\\t": b"\t",  # \t = tab
+    b"\\n": b"\n",  # \n = newline
```
Comment (Akuli) [inline porcupine/plugins/git_status.py:44]: Also `\r` could come up

Comment (Akuli) [reply porcupine/plugins/git_status.py:44]:
```suggestion
    b"\\r": b"\r",  # \r = CR byte (part of CRLF newline: \r\n)
    b"\\n": b"\n",  # \n = newline
```

File: tests/test_git_status_plugin.py [hunk 4]
```diff
@@ -29,6 +31,36 @@ def test_added_and_modified_content(tree, tmp_path, monkeypatch):
     assert set(tree.item(project_id, "tags")) == {"git_added"}
 
 
+weird_filenames = ["foo bar.txt", "foo'bar.txt", "örkkimörkkiäinen.ö", "bigyó.txt", "2π.txt"]
```
Comment (Akuli) [inline tests/test_git_status_plugin.py:34]:
```suggestion
weird_filenames = ["foo bar.txt", "foo'bar.txt", "foo\rbar.txt", "örkkimörkkiäinen.ö", "bigyó.txt", "2π.txt"]
```

File: tests/test_git_status_plugin.py [hunk 5]
```diff
@@ -29,6 +31,43 @@ def test_added_and_modified_content(tree, tmp_path, monkeypatch):
...
+        "foo:bar.txt",
+        'foo"bar.txt',
+        r"foo\bar.txt",
+        r"foo\123.txt",  # not a special escape code, only looks like it
+        "foo|bar.txt",
+        "foo?bar.txt",
+        "foo*bar.txt",
+        # Not mentioned in linked stackoverflow answer but still doesn't work on Windows
+        "foo\tbar.txt",
+        "foo\nbar.txt",
```
Comment (Akuli) [inline tests/test_git_status_plugin.py:56]:
```suggestion
        "foo\nbar.txt",
        "foo\rbar.txt",
```

File: tests/test_git_status_plugin.py [hunk 6]
```diff
@@ -29,6 +31,43 @@ def test_added_and_modified_content(tree, tmp_path, monkeypatch):
     assert set(tree.item(project_id, "tags")) == {"git_added"}
 
 
+weird_filenames = [
+    "foo bar.txt",
+    "foo'bar.txt",
+    "foo\rbar.txt",
```
Comment (Akuli) [inline tests/test_git_status_plugin.py:37]:
```suggestion
```

**Comment by Moosems** [comment]
Cool fix 😄.

**Pull request merged**